src/containers/Map/MapScreen.tsx

```diff
@@ -50,36 +54,95 @@ const MapScreen = () => {
 	const [showMapNodes, setShowMapNodes] = useState(false);
 
 	// Bot that was selected to send to some map node, used when showing map nodes
+	const botS: MarkerData | null = botSelected ? botSelected : null;
 	const [
 		selectedBotForOrder,
 		setSelectedBotForOrder,
-	] = useState<MarkerData | null>(null);
+	] = useState<MarkerData | null>(botS);
 
 	const [hasLocationPermission, setLocationPermission] = useState('null');
 	const [alert, setAlert] = useState(false);
 
 	const [loading, setLoading] = useState<boolean>(false);
 	const { state } = useContext(Ctx);
 
+	async function addToRoute(marker: MarkerData) {
+		console.log('adding marker ' + marker.name);
+		let curRoute = botRoute ? botRoute : [];
+		if (curRoute.indexOf(marker) > -1) {
+			await removeFromRoute(marker);
+		} else {
+			if (curRoute.length === 0 && selectedBotForOrder) {
+				let curPaths = paths ? paths : [];
+				let newPath = await MapService.getPathBetween(
+					selectedBotForOrder.location,
+					marker.location
+				);
+				curPaths.push(newPath);
```

Comment:
React nitpick, generally React suggests avoiding using in-place array modification methods like `push, splice` on state.
ie. you never want to mutate the state directly, even if you call `setState` on it after.

Here, `curPaths` refers to the same `paths` object, so if we then run `setPaths(curPaths)` after modifying the array in-place, there can be some undesired side effects

Instead they recommend to use modifications methods that create new arrays for example `concat` for inserts,
and `slice, filter` for removing elements like [here](https://stackoverflow.com/a/36326808).